hotel/src/crypto/aes.rs

```diff
@@ -0,0 +1,155 @@
+use core::cell::Cell;
+use hil::aes::{self, AesClient, Interrupt, AesModule, ParsedInterrupt};
+use hil::common::SyscallError;
+use super::keymgr::{KEYMGR0_REGS, Registers};
+
+pub struct AesEngine {
+    regs: *mut Registers,
+    client: Cell<Option<&'static AesClient>>,
+}
+
+impl AesEngine {
+    const unsafe fn new(regs: *mut Registers) -> AesEngine {
+        AesEngine {
+            regs: regs,
+            client: Cell::new(None),
+        }
+    }
+
+    pub fn set_client(&self, client: &'static AesClient) {
+        self.client.set(Some(client));
+    }
+
+    pub fn setup(&self, key_size: aes::KeySize, key: &[u8]) -> Result<isize, SyscallError> {
+        let ref regs = unsafe { &*self.regs }.aes;
+
+        self.enable_all_interrupts();
+        regs.ctrl.set(regs.ctrl.get() | key_size as u32 | AesModule::Enable as u32);
+        let mut i = 0;
+        for word in key.chunks(4) {
+            if i >= 8 {
+                println!("Input key too long!");
+                return Err(SyscallError::InvalidState);
+            }
+            let d = word.iter()
+                .map(|b| *b as u32)
+                .enumerate()
+                .fold(0, |accm, (i, byte)| accm | (byte << (i * 8)));
```

Comment:
It's certainly easy to add a crate dependency to the chip crate, in this case. FWIW, the byteorder package panics quite a lot in cases where the code above would "fail gracefully".

An important difference between the code above and @yuriks's suggestion is that the code above works transparently if the length of `key` is not divisible by 4.